### PR TITLE
add initial Mattermost-operator to do marketplace

### PR DESCRIPTION
**Pull request by cpanato**
Add initial mattermost-operator stack to DO mktplace

### CODE REVIEW

File: stacks/mattermost-operator/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# create minio-operator namespace
+kubectl create ns minio-operator
+# deploy minio-operator
+kubectl apply -n minio-operator -f https://raw.githubusercontent.com/mattermost/mattermost-operator/v0.7.0/docs/minio-operator/minio-operator.yaml
+
+# create mattermost-operator namespace
+kubectl create ns mattermost-operator
+# deploy mattermost-operator
+$ kubectl apply -n mattermost-operator -f https://raw.githubusercontent.com/mattermost/mattermost-operator/v0.7.0/docs/mattermost-operator/mattermost-operator.yaml
```
Comment (wadenick) [inline stacks/mattermost-operator/deploy.sh]: same errant `$` here

**Comment by arosales** [comment]
@cpanato Welcome to the DigitalOcean Marketplace. Thanks for the work to submit Mattemost. @wadenick has some feedback about a errent `$`.

**Comment by wadenick** [comment]
Hi @cpanato, 

I was able to get a trial installation of Mattermost Enterprise up and running with the following instructions. We'll likely need to include this (or similar) as the Getting Started guide in the listing, unless you have something better? Lastly, I think the only other thing required is the small fixes to the deploy scripts and this will be good to list in time for Kubecon! 

---- 

## Getting Started after Installing Mattermost Operator

After you have downloaded your kube config file, and are able to successfully connect 
to your DigitalOcean Kubernetes cluster (see https://cloud.digitalocean.com/kubernetes/clusters/ 
if you haven’t connected to your cluster) follow the instructions below to start Mattermost.

#### NOTE: In order to operate properly, your Mattermost cluster requires at least 3 nodes each with 8GB of RAM and 4vCPUs.

### Create your Mattermost Enterprise license secret

If you don't have a Mattermost Enterprise license you can register for a trial license 
at https://mattermost.com/trial/

Create a file named `mattermost-license-secret.yaml` with the following content:

```
apiVersion: v1
kind: Secret
metadata:
  name: mattermost-license
type: Opqaue
stringData:
  license: %LICENSE_FILE_CONTENTS%
```

replacing `%LICENSE_FILE_CONTENTS% with the content of your Mattermost license.

Apply it to your Mattermost cluster with: 

```
kubectl apply -f mattermost-license-secret.yaml
```

### Create your Mattermost Installation manifest and apply it to the cluster

You can create a trial Mattermost cluster quickly and easily by using the following settings, 
or alternatively if you have customized settings for your production deploy you can use those. 

Create a file named `mattermost-installation.yaml` with the following content:

```
apiVersion: mattermost.com/v1alpha1
kind: ClusterInstallation
metadata:
  name: mm-example-full
spec:
  size: 5000users
  image: mattermost/mattermost-enterprise-edition
  ingressName: mattermost.example.com
  useServiceLoadBalancer: true
  version: 5.15.0
  mattermostLicenseSecret: "mattermost-license"
  database:
    storageSize: 50Gi
  minio:
    storageSize: 50Gi
  elasticSearch:
    host: ""
    username: ""
    password: ""
```

You can read about customizations to your manifest at https://docs.mattermost.com/install/install-kubernetes.html#deploy-a-mattermost-installation

Apply it to your Mattermost cluster with:

```
kubectl create ns mattermost
```

followed by

```
kubectl apply -n mattermost -f mattermost-installation.yaml
```

Wait five minutes for the installation to complete, and then grab your LoadBalancer 
External IP with the following:

```
kubectl get services
```

### Connect to Mattermost and get going

You'll see output similar to the following, and can visit the LoadBalancer EXTERNAL-IP in 
your browser in order to get started with Mattermost:

```
NAME                           TYPE           CLUSTER-IP       EXTERNAL-IP       PORT(S)                      AGE
db-mysql                       ClusterIP      10.245.167.219   <none>            3306/TCP                     34m
db-mysql-master                ClusterIP      10.245.232.169   <none>            3306/TCP                     34m
kubernetes                     ClusterIP      10.245.0.1       <none>            443/TCP                      40m
mm-example-full                LoadBalancer   10.245.121.116   138.197.235.132   80:30265/TCP,443:32442/TCP   34m
mm-example-full-minio-hl-svc   ClusterIP      None             <none>            9000/TCP                     34m
mysql                          ClusterIP      None             <none>            3306/TCP,9125/TCP            34m
```

You can read more at https://docs.mattermost.com/install/install-kubernetes.html, including 
resource scaling guidelines for the number of users your installation will support.

**Comment by wadenick** [comment]
Hi @cpanato - I've removed the $ command prompt char's accidentally pasted in, and added a generic Getting Started section that works well on DigitalOcean with our LoadBalancer instead of Nginx Ingress. 

This is ready for approval and release IMO. Let us know if you have any further inputs or if we should go ahead - the goal is to have it in time for Kubecon. -nick

File: stacks/mattermost-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+# Mattermost Operator Stack
+
+This stack deploy the Mattermost Operator and the dependencies that are needed.
```
Comment (displague) [inline stacks/mattermost-operator/README.md]: "deploys"

File: stacks/mattermost-operator/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mattermost-license
+type: Opqaue
+stringData:
+  license: %LICENSE_FILE_CONTENTS%
+```
+
+replacing `%LICENSE_FILE_CONTENTS% with the content of your Mattermost license.
```
Comment (displague) [inline stacks/mattermost-operator/README.md]: missing "`"

**Comment by cpanato** [comment]
@wadenick thanks for updating the PR, i'm quite busy this week :/

**Comment by wadenick** [comment]
> @wadenick thanks for updating the PR, i'm quite busy this week :/

No problem, we gotchu @cpanato 



> Although I'm not connected to DO or the marketplace project, I am generally interested in the marketplace and the Mattermost installation process to help me understand what it would take to add http://crossplane.io/.
> 
> I found myself reading through the README here so I pointed out a few typos I encountered.

Thanks for the typo pointers @displague, and we’d love to talk about crossplane.io, can you catch us at Kubecon?

**Comment by paulanunda** [comment]
@cpanato @wadenick @arosales @displague I opened up a new PR based on this one that adds nginx-ingress into the deployment -- https://github.com/digitalocean/marketplace-kubernetes/pull/62. If it looks good, I think we can close this, get #62 merged, and listed in the marketplace.

**Comment by paulanunda** [comment]
After making a few changes (added nginx-ingress to deploy.sh), I was able to get it to work w/o issue:

```
Desktop ❯ k get po
NAME                               READY   STATUS    RESTARTS   AGE
db-mysql-0                         4/4     Running   0          11m
db-mysql-1                         4/4     Running   0          8m38s
db-mysql-2                         4/4     Running   0          6m45s
mm-example-full-7bf5599894-7xsff   1/1     Running   0          11m
mm-example-full-7bf5599894-f5vrm   1/1     Running   2          11m
mm-example-full-minio-0            1/1     Running   0          11m
mm-example-full-minio-1            1/1     Running   0          11m
mm-example-full-minio-2            1/1     Running   0          11m
mm-example-full-minio-3            1/1     Running   0          11m
```

```
Desktop ❯ k get ingress
NAME              HOSTS                      ADDRESS         PORTS   AGE
mm-example-full   mm-test-1.paulanunda.com   157.230.65.59   80      11m
```

image

**Pull request merged**